pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -262,6 +263,23 @@ func (pl *VolumeBinding) Filter(ctx context.Context, cs *framework.CycleState, p
 	return nil
 }
 
+// PreScore invoked at the preScore extension point. It checks whether volumeBinding can skip Score
+func (pl *VolumeBinding) PreScore(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) *framework.Status {
+	if pl.scorer == nil {
+		return framework.NewStatus(framework.Skip)
+	}
+	state, err := getStateData(cs)
+	if err != nil {
+		return framework.AsStatus(err)
+	}
+	for _, podVolumes := range state.podVolumesByNode {
```

Comment:
This is going to be iterating over all nodes in a non-parallel loop, which might be more expensive than just running Score. So I'm not sure if we are gaining anything.

I would prefer a simpler check, like `podVolumesByNode` being empty.

Reply:
How about we add a bool flag in state data?
```
type stateData struct {
	allBound bool
	podVolumesByNode  map[string]*PodVolumes
	podVolumeClaims   *PodVolumeClaims
	sync.Mutex
    
    // hasStaticBindings declares whether the pod contains one or more StaticBinding. 
	// If not, vloumeBinding will skip score extension point.
	hasStaticBindings bool
}
```

Reply:
I think so, you can get the boolean in Filter.